funnel/models/moderation.py

```diff
@@ -47,16 +53,24 @@ def get_one(cls, exclude_user=None):
 
         return reports.order_by(db.func.random()).first()
 
-
-def _report_comment(self, actor):
-    report = CommentModeratorReport.query.filter_by(
-        user=actor, comment=self
-    ).one_or_none()
-    if report is None:
-        report = CommentModeratorReport(user=actor, comment=self)
-        db.session.add(report)
-        db.session.commit()
-    return report
+    @with_roles(grants={'site_editor'})
+    @property
+    def users_who_are_site_editors(self):
+        return User.query.join(
+            SiteMembership, SiteMembership.user_id == User.id
+        ).filter(
+            SiteMembership.is_active.is_(True), SiteMembership.is_site_editor.is_(True)
+        )
 
 
-Comment.report_spam = _report_comment
+@reopen(Comment)
+class Comment:
+    def report_spam(self, actor):
+        report = CommentModeratorReport.query.filter_by(
+            user=actor, comment=self
+        ).one_or_none()
+        if report is None:
+            report = CommentModeratorReport(user=actor, comment=self)
```

Comment:
Make this `CommentModeratorReport.submit(...)` and let that class deal with the deduplication.